components/properties/src/props.rs

```diff
@@ -41,6 +49,147 @@ enum EnumeratedProperty {
     InvalidCode = -1, // TODO(#1160) - taken from ICU4C UProperty::UCHAR_INVALID_CODE
 }
 
+/// Private marker type for PropertyValueLookup
+/// to work for all properties at once
+#[derive(Clone, Copy, PartialEq, Eq, Hash, Debug)]
+pub(crate) struct ErasedValueNameMap;
+impl DataMarker for ErasedValueNameMap {
+    type Yokeable = PropertyValueNameMapV1<'static>;
+}
+
+/// A struct capable of looking up a property value from a string name
+///
+/// The name can be a short name (`Lu`), a long name(`Uppercase_Letter`),
+/// or an alias.
+///
+/// Property names can be looked up using "strict" matching (looking for a name
+/// that matches exactly), or "loose matching", where the name is allowed to deviate
+/// in terms of ASCII casing, whitespace, underscores, and hyphens.
+///
+/// # Example
+///
+/// ```
+/// use icu::properties::GeneralCategory;
+///
+/// let lookup = GeneralCategory::lookup_values(&icu_testdata::unstable())
+///                  .expect("The data should be valid");
+/// // short name for value
+/// assert_eq!(lookup.get_strict("Lu"), Some(GeneralCategory::UppercaseLetter));
+/// assert_eq!(lookup.get_strict("Pd"), Some(GeneralCategory::DashPunctuation));
+/// // long name for value
+/// assert_eq!(lookup.get_strict("Uppercase_Letter"), Some(GeneralCategory::UppercaseLetter));
+/// assert_eq!(lookup.get_strict("Dash_Punctuation"), Some(GeneralCategory::DashPunctuation));
+/// // name has incorrect casing
+/// assert_eq!(lookup.get_strict("dashpunctuation"), None);
+/// // loose matching of name
+/// assert_eq!(lookup.get_loose("dash-punctuation"), Some(GeneralCategory::DashPunctuation));
+/// // fake property
+/// assert_eq!(lookup.get_strict("Animated_Gif"), None);
+/// ```
+pub struct PropertyValueLookup<T> {
+    map: DataPayload<ErasedValueNameMap>,
+    marker: PhantomData<fn() -> T>,
+}
+
+impl<T: TrieValue> PropertyValueLookup<T> {
+    /// Get the property value as a u16, doing a strict search looking for
+    /// names that match exactly
+    #[inline]
+    pub fn get_strict_u16(&self, name: &str) -> Option<u16> {
```

Comment:
I've added u16 methods since FFI will benefit from not needing to go through the concrete type

Reply:
These make sense for FFI, but do we want to lose the type safety on the Rust side? Maybe `#[doc(hidden)]` these?

Reply:
We have `get_u32()` etc on other mappy types already. It's a somewhat common pattern.